src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsalertsapi/resource/RetrieveAlertIntTest.kt

```diff
@@ -0,0 +1,205 @@
+  fun `404 alert not found`() {
+    val response = webTestClient.get()
+      .uri("/alerts/$uuid")
+      .headers(setAuthorisation(roles = listOf(ROLE_ALERTS_READER)))
+      .headers(setAlertRequestContext())
+      .exchange()
+      .expectStatus().isNotFound
+      .expectBody(ErrorResponse::class.java)
+      .returnResult().responseBody
+
+    with(response!!) {
+      assertThat(status).isEqualTo(404)
+      assertThat(errorCode).isNull()
+      assertThat(userMessage).isEqualTo("Alert not found: Could not find alert with uuid $uuid")
+      assertThat(developerMessage)
+        .isEqualTo("Could not find alert with uuid $uuid")
+      assertThat(moreInfo).isNull()
+    }
+  }
+
+  @Test
+  fun `retrieve alert`() {
+    val alert = createAlert()
+    val response = webTestClient.get()
+      .uri("/alerts/${alert.alertUuid}")
+      .headers(setAuthorisation(roles = listOf(ROLE_ALERTS_READER)))
+      .headers(setAlertRequestContext())
+      .exchange()
+      .expectStatus().isOk
+      .expectBody(Alert::class.java)
+      .returnResult().responseBody
+    val alertEntity = alertRepository.findByAlertUuid(alert.alertUuid)!!
+    val alertCode = alertCodeRepository.findByCode(alertEntity.alertCode.code)!!
+
+    with(response!!) {
+      assertThat(alertEntity).usingRecursiveAssertion().ignoringFields("auditEvents").isEqualTo(
+        uk.gov.justice.digital.hmpps.hmppsalertsapi.entity.Alert(
+          1,
+          alert.alertUuid,
+          alertCode,
+          prisonNumber,
+          description,
+          authorisedBy,
+          activeFrom!!,
+          activeTo,
+        ),
+      )
+    }
+
+    with(alertEntity.auditEvents()[0]) {
+      assertThat(auditEventId).isEqualTo(1)
+      assertThat(action).isEqualTo(AuditEventAction.CREATED)
+      assertThat(description).isEqualTo(
+        "Alert created",
+      )
+      assertThat(actionedAt).isCloseToUtcNow(within(3, ChronoUnit.SECONDS))
+      assertThat(actionedBy).isEqualTo(TEST_USER)
+      assertThat(actionedByDisplayName).isEqualTo(TEST_USER_NAME)
+    }
+  }
+
+  private fun createAlertRequest(
+    prisonNumber: String = PRISON_NUMBER,
+    alertCode: String = ALERT_CODE_VICTIM,
+  ) =
+    CreateAlert(
+      prisonNumber = prisonNumber,
+      alertCode = alertCode,
+      description = "Alert description",
+      authorisedBy = "A. Authorizer",
+      activeFrom = LocalDate.now().minusDays(3),
+      activeTo = null,
+    )
+
+  private fun createAlert(): Alert {
+    val request = createAlertRequest()
+    return webTestClient.post()
+      .uri("/alerts")
+      .bodyValue(request)
+      .headers(setAuthorisation(user = TEST_USER, roles = listOf(ROLE_ALERTS_WRITER), isUserToken = true))
+      .exchange()
+      .expectStatus().isCreated
+      .expectHeader().contentType(MediaType.APPLICATION_JSON)
+      .expectBody(Alert::class.java)
+      .returnResult().responseBody!!
+  }
+  private fun updateAlertRequest(comment: String = "Another update alert") =
```

Comment:
I don't think you need this update request?

Reply:
🤦‍♂️ D'oh - no!